fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowEncryptKeysCommand.java

```diff
@@ -0,0 +1,134 @@
+    private static final ShowResultSetMetaData META_DATA =
+            ShowResultSetMetaData.builder()
+                    .addColumn(new Column("EncryptKey Name", ScalarType.createVarchar(20)))
+                    .addColumn(new Column("EncryptKey String", ScalarType.createVarchar(1024)))
+                    .build();
+
+    private String dbName;
+    private final String wild;
+
+    public ShowEncryptKeysCommand(String databaseName, String wildWhere) {
+        super(PlanType.SHOW_ENCRYPT_KEYS_COMMAND);
+        this.dbName = databaseName;
+        this.wild = wildWhere;
+    }
+
+    private void validate(ConnectContext ctx) throws AnalysisException {
+        if (Strings.isNullOrEmpty(dbName)) {
+            dbName = ctx.getDatabase();
+            if (Strings.isNullOrEmpty(dbName)) {
+                ErrorReport.reportAnalysisException(ErrorCode.ERR_NO_DB_ERROR);
+            }
+        }
+
+        // check auth
+        if (!Env.getCurrentEnv().getAccessManager().checkGlobalPriv(ConnectContext.get(), PrivPredicate.ADMIN)) {
+            ErrorReport.reportAnalysisException(ErrorCode.ERR_SPECIFIC_ACCESS_DENIED_ERROR,
+                    PrivPredicate.ADMIN.getPrivs().toString());
+        }
+    }
+
+    @Override
+    public ShowResultSet doRun(ConnectContext ctx, StmtExecutor executor) throws Exception {
+        validate(ctx);
+        Util.prohibitExternalCatalog(ctx.getDefaultCatalog(), this.getClass().getSimpleName());
+        DatabaseIf db = ctx.getCurrentCatalog().getDbOrAnalysisException(dbName);
+        List<List<String>> resultRowSet = Lists.newArrayList();
+        if (db instanceof Database) {
+            List<EncryptKey> encryptKeys = ((Database) db).getEncryptKeys();
+            List<List<Comparable>> rowSet = Lists.newArrayList();
+            for (EncryptKey encryptKey : encryptKeys) {
+                List<Comparable> row = encryptKey.getInfo();
+                // like predicate
+                if (wild == null || like(encryptKey.getEncryptKeyName().getKeyName())) {
+                    rowSet.add(row);
+                }
+
+            }
+
+            // sort function rows by first column asc
+            ListComparator<List<Comparable>> comparator = null;
+            OrderByPair orderByPair = new OrderByPair(0, false);
+            comparator = new ListComparator<>(orderByPair);
+            Collections.sort(rowSet, comparator);
+
+            Set<String> encryptKeyNameSet = new HashSet<>();
+            for (List<Comparable> row : rowSet) {
+                List<String> resultRow = Lists.newArrayList();
+                for (Comparable column : row) {
+                    resultRow.add(column.toString());
+                }
+                resultRowSet.add(resultRow);
+                encryptKeyNameSet.add(resultRow.get(0));
+            }
+        }
+
+        return new ShowResultSet(META_DATA, resultRowSet);
+    }
+
+    public boolean like(String str) {
```

Comment:
```suggestion
    private boolean like(String str) {
```
better to change it as private method, you can do it in your next other pr :smile: